docs/source/geo_def.rst

```diff
@@ -88,10 +89,83 @@ based on area extent and a proj4-string or a list of proj4 arguments.
+
+where
+
+* **x_ul**: projection x coordinate of upper left corner of upper left pixel
+* **y_ul**: projection y coordinate of upper left corner of upper left pixel
+* **center_x**: projection x coordinate of center of projection
+* **center_y**: projection y coordinate of center of projection
+* **height**: number of pixels in y direction (number of grid rows)
+* **width**: number of pixels in x direction (number of grid columns)
+* **x_size**: projection size of pixels in the x direction
+* **y_size**: projection size of pixels in the y direction
+* **x_length**: projection length from center to left/right outer edge
+* **y_length**: projection length from center to top/bottom outer edge
+* **units** accepts anything with 'm', 'rad', 'deg' or '°'. The order of default is: units expressed
+  with each variable, units passed to **units**, units used in **projection**, then meters. **shape** is not affected
+  by units. To add units to a specific variable, you can make an xarray.DataArray with a variable and then use
+  {'units': unit_of_choice} as its attrs attribute.
+
+**from_params** returns the following outcomes:
+
+1. If shape and area_extent are found, an AreaDefinition is returned
+2. If only shape or area_extent can be found, a DynamicAreaDefinition is returned
+3. If neither shape nor area_extent can be found, an error will occur
+
+.. doctest::
+
+ >>> from pyresample import utils
+ >>> from xarray import DataArray
+ >>> area_id = 'ease_sh'
+ >>> description = 'Antarctic EASE grid'
+ >>> proj_id = 'ease_sh'
+ >>> projection = {'a': '6371228.0', 'units': 'm', 'lon_0': '0', 'proj': 'laea', 'lat_0': '-90'}
+ >>> area_extent = DataArray((-135.0, -17.516001139327766, 45.0, -17.516001139327766), attrs={'units': 'degrees'})
+ >>> pixel_size = 25067.525
+ >>> center = [0, 0]
+ >>> area_def = utils.from_params(description, proj_dict, area_extent=area_extent, center=center, pixel_size=pixel_size,
+ ...                              area_id=area_id, proj_id=proj_id)
+ >>> print(area_def)
+ Area ID: ease_sh
+ Description: Antarctic EASE grid
+ Projection ID: ease_sh
+ Projection: {'a': '6371228.0', 'lat_0': '-90.0', 'lon_0': '0.0', 'proj': 'laea', 'units': 'm'}
+ Number of columns: 425
+ Number of rows: 425
+ Area extent: (-5326849.0625, -5326849.0625, 5326849.0625, 5326849.0625)
+
+The **load_area** function can be used to parse area definitions from a configuration file by giving it the area file
+name and regions you wish to load. **load_area** takes advantage of **from_params** and hence uses the same arguments.
+
+When composing a yaml file, things to keep in mind are:
+
+* If the elements of **shape**, **pixel_size**, or **radius** respectively are the same, you may define the
+  variable by the element alone. Example: [0, 0] = 0.
+* **area_id** defaults to the name used to specify the area definition (in this case ease_sh or ease_nh).
+* You may add a key to a variable with the same name as the variable itself. This variable-key has the
```

Comment:
This is extremely confusing. Maybe leave it to examples below to describe these situations by showing instead of telling (I don't think you even need to describe it with words). This entire set of "keep in mind" is probably best as a show rather than a tell.